napari/napari
# Fixes several sphinx warnings.

# Description
This fixes several straightforward Sphinx warnings and simplifies the docs build. 

Before: 384 warnings
Now: 290 warnings

## Type of change


# How has this been tested?

Locally, you can do `make docs` to see the build with these fixes. In addition, if you want to confirm that this works when deployed to napari.github.io, you can:
1. Clone both [napari/napari](https://github.com/napari/napari) and [napari/napari.github.io](https://github.com/napari/napari.github.io)
2. Go into the napari/napari folder and run
   ```bash
   $ pip install -e '.[all]'
   $ make docs
   ```
3. View changes rendered in "_build/html"
4. Run `python docs/copy-docs.py` (this will copy the appropriate sources into the napari/napari.github.io folder and update the `napari/napari.github.io/_toc.yml` file
5. Go into the napari/napari.github.io folder and run
	```bash
	$ yarn dev
	```
6. Confirm that this command renders the appropriate outputs.

### A few things to note
- Check out the difference between https://napari.org/api/stable/napari.Viewer.html and this build (http://localhost:8080/api/stable/napari.Viewer.html):

**Old**
![napariold](https://user-images.githubusercontent.com/3949932/143780242-3f89ef2a-bce8-46ec-af31-8924b2a1a63f.png)

**New**
![naparinew](https://user-images.githubusercontent.com/3949932/143780767-937c0bc9-4d48-493a-83d8-0aa036bacd72.png)
![naparinew2](https://user-images.githubusercontent.com/3949932/143780766-9cfbdf3c-d0f4-4cb6-9723-375e973382fb.png)

- Currently, there is a page that appears twice in the ToC (`api/stable/napari.html`). This is due to sphinx not being able to process a `:hidden:` option for the autodoc directive and I'm looking for solutions.
- The remaining warnings are mostly of two types: Sphinx not being able to parse typehints, and repeated references to autogenerated docs. Both should be fixed in a later PR.
 
## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).
